projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/listeners/Function.java

```diff
@@ -982,6 +982,20 @@ private static int getAlign(long argument) {
         return (int) argument & (Long.SIZE - 1);
     }
 
+    private static Type getExtractedElementType(Type type) {
+        Type valType = type;
+        if (valType instanceof PointerType) {
```

Comment:
I completely overlooked the `types.get(i++)`, thanks for noticing! `i` is only the index in the argument array and will only ever retrieve the correct type if it happens to be equal to `args[i]`. This same mistake happens in several other places too.